testing/junit/src/main/java/com/linecorp/armeria/testing/junit/server/mockwebserver/MockResponse.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit.server.mockwebserver;
+
+import java.time.Duration;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+
+/**
+ * A mock response to return to a request made to {@link MockWebServerExtension}.
+ */
+public class MockResponse {
```

Comment:
How aboug writing an `HttpResponse` implementation that defers the publication, which may be useful for non-testing scenarios as well, e.g. `HttpResponse.delayed(HttpResponse, Duration)` or `delayed(Supplier<HttpResponse>, Duration)`.

Reply:
Good idea - went with that and removed `MockResponse` for now since if we don't have any parameters besides delay, there's no point for it. If we need a test-only interface later can add it back.